generic_test.go

```diff
@@ -30,7 +28,11 @@ func TestGenericDisconnect(t *testing.T) {
 	defer stream.Stop()
 
 	w := httptest.NewRecorder()
-	assert.Equal(t, resyncErr, stream.Subscribe(w, "first"))
+	err := stream.Subscribe(w, "first")
+
+	if !errors.Is(err, resyncErr) {
+		t.Errorf("Expected error %v, got %v", resyncErr, err)
+	}
```

Comment:
Keep error declaration and checks together
```suggestion
	err := stream.Subscribe(w, "first")
	if !errors.Is(err, resyncErr) {
		t.Errorf("Expected error %v, got %v", resyncErr, err)
	}
```